andela/boondocks-bn-backend
# #169257404 Make a return trip feature

#### What does this PR do?
Implement a feature to make return trip request

#### Description of Task to be completed?
- [x] Creates a booking, rooms, hotel, trips models
- [x] Creates a return trip
- [x] Allows a user to book accommodation

#### How should this be manually tested?
- Clone the repo and do `npm install` to install dependencies
- Run `npx sequelize-cli db:migrate` or `sequelize db:migrate` depending on your configurations to create required database tables
- Run `npm run watch`
- Run `npm run pretest-travis` and then `npm test` to run tests
- Using postman, make a trip using `POST /api/v1/trips/return`
```
// Sample data
{
      "hotelId": 1,
      "type":"one way",
      "leavingFrom": "Nairobi",
      "goingTo": "Madrid",
      "travelDate": "2019-11-09",
      "reason": "business",
      "rooms": [15, 21]
}
```
- Visit `localhost:<PORT>/api/docs` in to check documentation for endpoint

#### Any background context you want to provide?
`Rooms` field is an array of roomId booked by the user.
#### What are the relevant pivotal tracker stories?
[#169257404
](https://www.pivotaltracker.com/story/show/169257404)

#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/15703806/69320818-93806580-0c4a-11ea-9664-08eb9546abb0.png)

#### Questions:
N/A
